src/pm.rs

```diff
@@ -15,7 +16,53 @@ const SOURCE_PATH: &str = "/etc/apt/sources.list.d/atm.list";
 const STATE_PATH: &str = "/var/lib/atm/state";
 const STATE_DIR: &str = "/var/lib/atm/";
 const DPKG_STATE: &str = "/var/lib/dpkg/status";
-const MIRROR_URL: &str = "https://repo.aosc.io/debs";
+
+#[derive(Deserialize, Debug)]
+struct AptGenListStatus {
+    mirror: Vec<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct Mirror {
+    url: String,
+}
+
+#[derive(Deserialize, Debug)]
+struct Mirrors {
+    mirror: HashMap<String, Mirror>,
+}
+
+pub fn get_mirror_url() -> Result<String> {
+    let mut status_file = fs::File::open("/var/lib/apt/gen/status.json")?;
+    let mut status_data = String::new();
+    status_file.read_to_string(&mut status_data)?;
+
+    let status_v: AptGenListStatus = serde_json::from_str(&status_data)?;
+    let mut mirror = String::from("origin");
+    if let Some(m) = &status_v.mirror.get(0) {
+        mirror = m.to_string();
+    }
```

Comment:
You can actually do this:

```suggestion
    let mirror;
    if let Some(m) = &status_v.mirror.get(0) {
        mirror = m.to_string();
    } else {
        mirror = "origin".to_string();
    }
```

... to avoid using `mut`.